### PR TITLE
Create message-driven pub/sub sample

### PR DESCRIPTION
**Pull request by timbussmann**
Create a dedicated sample using message-driven pub/sub to better document the difference between the two approaches and also make the individual sample's description simpler by focusing on one pub/sub mechanism.

### CODE REVIEW

File: samples/pubsub/message-driven/Core_6/Publisher/Program.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        var endpointConfiguration = new EndpointConfiguration("Samples.PubSub.MessageDrivenPublisher");
+        endpointConfiguration.UsePersistence<InMemoryPersistence>();
+        endpointConfiguration.UseTransport<MsmqTransport>();
+
+        endpointConfiguration.SendFailedMessagesTo("error");
+        endpointConfiguration.EnableInstallers();
+
+        var endpointInstance = await Endpoint.Start(endpointConfiguration)
+            .ConfigureAwait(false);
+        await Start(endpointInstance)
```
Comment (andreasohlund) [inline samples/pubsub/message-driven/Core_6/Publisher/Program.cs:19]: The name start feels confusing that there is a start call on the line above. How about `PublishOrderEvent` or something?

File: samples/pubsub/message-driven/Core_7/Publisher/Program.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        var endpointConfiguration = new EndpointConfiguration("Samples.PubSub.MessageDrivenPublisher");
+        endpointConfiguration.UsePersistence<InMemoryPersistence>();
+        endpointConfiguration.UseTransport<MsmqTransport>();
+
+        endpointConfiguration.SendFailedMessagesTo("error");
+        endpointConfiguration.EnableInstallers();
+
+        var endpointInstance = await Endpoint.Start(endpointConfiguration)
+            .ConfigureAwait(false);
+        await Start(endpointInstance)
```
Comment (andreasohlund) [inline samples/pubsub/message-driven/Core_7/Publisher/Program.cs:19]: Same comment as above

**Review by mauroservienti — APPROVED** [review]

File: samples/pubsub/message-driven/sample.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+---
+title: Message-driven Publish/Subscribe
+summary: Persistence based Publish/Subscribe for unicast-only transports.
+reviewed: 2020-09-28
+component: Core
+related:
+ - nservicebus/messaging/publish-subscribe
+---
+
+This sample shows how to publish an event message from a messaging endpoint in one physical process, subscribe to the event in a separate messaging endpoint in a second physical process, and execute a message handler when an event message is received. This sample uses the [message-driven publish-subscribe mechanism](/nservicebus/messaging/publish-subscribe#mechanics-message-driven-persistence-based) for unicast transports.
```
Comment (andreasohlund) [inline samples/pubsub/message-driven/sample.md:10]:
```suggestion
This sample shows how to publish an event from one endpoint, subscribe to the event in a separate endpoint and execute a message handler when an event is received. This sample uses the [message-driven publish-subscribe mechanism](/nservicebus/messaging/publish-subscribe#mechanics-message-driven-persistence-based) for unicast transports.
```

File: samples/pubsub/message-driven/sample.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+downloadbutton
+
+Before running the sample, look over the solution structure, the projects and the classes. The projects `Publisher` and `Subscriber` are console applications that each host an instance of an NServiceBus messaging endpoint.
+
+## Defining messages
+
+The `Shared` project contains the definition of the messages that are sent between the processes. Open "OrderReceived.cs" to see the message that will be published by this sample. Note that this event implements an interface called `IEvent` to denote that this message is an event. To define messages without adding a dependency to NServiceBus, use [Unobtrusive Mode Messages](/nservicebus/messaging/unobtrusive-mode.md). 
+
+## Creating and publishing messages
```
Comment (andreasohlund) [inline samples/pubsub/message-driven/sample.md:20]:
```suggestion
## Publishing the event
```

File: samples/pubsub/message-driven/sample.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+As the name implies, the `Publisher` project is a publisher of event messages. It uses the NServiceBus API to publish the `OrderReceived` event every time the `1` key is pressed. The created message is populated and [published](/nservicebus/messaging/publish-subscribe/) using the `Publish` API.
+
+snippet: PublishLoop
+
+## Implementing subscribers
```
Comment (andreasohlund) [inline samples/pubsub/message-driven/sample.md:26]:
```suggestion
## Subscribing to the event
```

File: samples/pubsub/message-driven/sample.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+To receive messages from the publisher, the subscribers [must subscribe to the message types](/nservicebus/messaging/publish-subscribe/) they are designed to handle. A subscriber must have a handler for the type of message and a configuration that tells the endpoint where to send subscriptions for messages to:
+
+snippet: SubscriptionConfiguration
+
+ * The `Subscriber` handles and subscribes to the `OrderReceived` type.
```
Comment (andreasohlund) [inline samples/pubsub/message-driven/sample.md:32]:
```suggestion
 * The `Subscriber` handles and subscribes to the `OrderReceived` event type.
```

File: samples/pubsub/message-driven/sample.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+ * The handlers in each project are in files that end with the word `Handler` for example `OrderReceivedHandler.cs`.
+ * `Subscriber` uses the default auto-subscription feature of the bus where the bus automatically subscribes to the configured publisher. [The auto-subscribe feature can be explicitly disabled](/nservicebus/messaging/publish-subscribe/controlling-what-is-subscribed.md) as part of the endpoint configuration.
+  
+## Run the sample
+
+When running the sample, notice the two open console applications. Bring the `Publisher` endpoint to the foreground.
```
Comment (andreasohlund) [inline samples/pubsub/message-driven/sample.md:38]:
```suggestion
When running the sample, two open console applications will open. Bring the `Publisher` endpoint to the foreground.
```

File: samples/pubsub/message-driven/sample.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+Click the `1` key repeatedly in the `Publisher` process console window and notice how the messages appear in the `Subscriber` console window.
+
+## Message Flow
```
Comment (andreasohlund) [inline samples/pubsub/message-driven/sample.md:42]:
```suggestion
## Subscription mechanics
```

File: samples/pubsub/native/sample.md [hunk 9]
```diff
@@ -43,12 +43,12 @@ Click the `1` key repeatedly in the `Publisher` process console window and notic
 
 ## Message Flow
 
-The exact message flow may differ between [unicast transports](/transports/types.md#unicast-only-transports) and [broker transports](/transports/types.md#broker-transports). See the [publish-subscribe documentation](/nservicebus/messaging/publish-subscribe/) for further details.
+In multicast-enabled transports, the broker handles the subscription mechanism, routing published events to all subscribed endpoints. See the [publish-subscribe documentation](/nservicebus/messaging/publish-subscribe/#mechanics-native) for further details.
```
Comment (andreasohlund) [inline samples/pubsub/native/sample.md:46]:
```suggestion
In multicast-capable transports, the broker handles the subscription mechanism, routing published events to all subscribed endpoints. See the [publish-subscribe documentation](/nservicebus/messaging/publish-subscribe/#mechanics-native) for further details.
```

**Review by andreasohlund — APPROVED** [review]
Mostly nitpicks

File: samples/pubsub/message-driven/Core_6/Publisher/Program.cs (hunk 1 above)
Comment (timbussmann) [reply samples/pubsub/message-driven/Core_6/Publisher/Program.cs:19]: fair point, took that from the existing sample but sounds like a good improvement 👍

File: samples/pubsub/message-driven/sample.md (hunk 7 above)
Comment (timbussmann) [reply samples/pubsub/message-driven/sample.md:38]:
```suggestion
When running the sample, two console applications will open. Bring the `Publisher` endpoint to the foreground.
```

File: samples/pubsub/native/sample.md (hunk 9 above)
Comment (timbussmann) [reply samples/pubsub/native/sample.md:46]: I took "multicast-enabled" from here: https://docs.particular.net/transports/types#multicast-enabled-transports, is it worth to maintain a consistent language?

**Review by andreasohlund — APPROVED** [review]

**Pull request merged**